src/components/injected/YTVideoPage.tsx

```diff
@@ -33,137 +28,130 @@ const useStyles = makeStyles(() => ({
-
-          observer.observe(document, { childList: true, subtree: true });
-
-          return () => {
-            observer.disconnect();
-          };
-        }, [currentVideoId]);
-
-        React.useEffect(() => {
-          if (!settings.active || !settings.enhanceYouTubeExperience) {
-            patchYTRecommendations(2);
-          }
-        }, [settings]);
-
-        // do not show this component when extension is not `active` or `ux` is disabled
-        if (!settings.active || !settings.enhanceYouTubeExperience) {
-          return null;
-        }
-
-        return (
-          <Box>
-            <AppBar className={classes.appBar} position="static">
-              <Tabs
-                value={currentTab}
-                onChange={(e, n) => onTabChange(n)}
-                aria-label="recommendations tabs"
-                variant="fullWidth"
-                centered
-              >
-                <Tab
-                  className={classes.tab}
-                  icon={<ContentCreatorIcon />}
-                  wrapped={true}
-                  label={t('creator:title')}
-                  index={0}
-                />
-                <Tab
-                  className={classes.tab}
-                  icon={<YTIcon />}
-                  wrapped={true}
-                  label={t('youtube:title')}
-                  index={1}
-                />
-                <Tab
-                  className={classes.tab}
-                  icon={<HideIcon />}
-                  wrapped={true}
-                  label={t('hide_all:title')}
-                  index={2}
-                />
-              </Tabs>
-            </AppBar>
-
-            <TabPanel value={currentTab} index={0}>
-              {currentVideoId === undefined ? (
-                <Box>
-                  <FullSizeLoader />
-                </Box>
-              ) : (
-                <VideoRecommendations
-                  queries={{
-                    videoRecommendations: { videoId: currentVideoId },
-                  }}
-                />
-              )}
-            </TabPanel>
-            <TabPanel value={currentTab} index={2}>
-              <Typography variant="h4">{t('common:empty_string')}</Typography>
-            </TabPanel>
-          </Box>
+    const observer = new MutationObserver(() => {
+      const newVideoId = getVideoId(document.location.href);
+      if (newVideoId !== currentVideoId) {
+        logger.debug(
+          'Video id changed from (%s) to (%s)',
+          currentVideoId,
+          newVideoId
         );
-      })}
-    />
-  );
+        setVideoId(newVideoId);
+      }
+    });
+
+    observer.observe(document, { childList: true, subtree: true });
+
+    return () => {
+      observer.disconnect();
+      patchYTRecommendations(YT_TAB_INDEX);
+    };
+  }, [currentVideoId]);
+
+  React.useEffect(() => {
+    if (!settings.active || !settings.enhanceYouTubeExperience) {
+      patchYTRecommendations(YT_TAB_INDEX);
```

Comment:
why do we run this if the switch is off? and what is settings.active? what turns it off?

Reply:
the `settings.active` prop was the one assigned to the "enable" button in the Popup, but we got rid of it.

but I need to patch the YT UI when the `settings.enhanceYoutubeExperience` get disabled to restore the YT recommendations tab

Reply:
understood. We'll have to do some cleaning of the config at some point I think